src/rules/no-assign-ref.ts

```diff
@@ -0,0 +1,38 @@
+import { createRule } from '../utils/create-rule';
+
+type MessageIds = 'noAssignRef';
+type Options = [];
+
+export const noAssignRef = createRule<MessageIds, Options>({
+    meta: {
+        type: 'suggestion',
+        fixable: 'code',
+        docs: {
+            description: 'Disallow assigning by reference',
+        },
+        messages: {
+            noAssignRef:
+                "Assigning by reference is not allowed.",
+        },
+        schema: [],
+    },
+
+    create(context) {
+        return {
+            "assignref"(node) {
+                context.report({
+                    node,
+                    messageId: 'noAssignRef',
+                    fix(fixer) {
+                        const nodeText = context.sourceCode.getText(node);
+
+                        return fixer.replaceText(
+                            node,
+                            nodeText.replace(/&/g, ''),
+                        );
+                    },
+                });
```

Comment:
This will autofix when running using `--fix`, and it'll change the code's meaning.
Use fixes only when it is safe to change the user's code, and [suggestions](https://eslint.org/docs/latest/extend/custom-rules#providing-suggestions) otherwise.